feature/session/src/main/java/io/github/droidkaigi/confsched2020/session/ui/viewmodel/SessionDetailViewModel.kt

```diff
@@ -76,21 +109,49 @@ class SessionDetailViewModel @AssistedInject constructor(
             }
         }
         val showEllipsis = descriptionTextExpandState == TextExpandState.COLLAPSED
+        val totalThumbsUpCount = when (totalThumbsUpCountLoadState) {
+            is LoadState.Loaded -> {
+                totalThumbsUpCountLoadState.value
+            }
+            else -> {
+                current.thumbsUpCount.total
+            }
+        }
+
+        val incrementThumbsUpCount = incrementThumbsUpCountResult.getOrDefault(0)
+
+        val thumbsUpCount = ThumbsUpCount(
+            total = totalThumbsUpCount,
+            incremented = incrementThumbsUpCount,
+            incrementedUpdated = current.thumbsUpCount.incremented != incrementThumbsUpCount
+        )
 
         UiModel(
             isLoading = isLoading,
-            error = sessionLoadState
-                .getErrorIfExists()
-                .toAppError()
-                ?: favoriteState
-                    .getErrorIfExists()
-                    .toAppError(),
             session = sessions,
             showEllipsis = showEllipsis,
-            searchQuery = searchQuery
+            searchQuery = searchQuery,
+            thumbsUpCount = thumbsUpCount
         )
     }
 
+    val appError: LiveData<AppError?> = merge(
```

Comment:
I have a question. Please tell me why you separated the error 🙋‍♂ 
for simplifying??

Reply:
Yes.
The reason is that uiModel's errors is complexed.
In my opinion, it is easy to read that the code is separated success and failure flow.

What do you think?

Reply:
You have added the errorGettable interface. 👍 
I think that you can write enough with UiModel by using this. 👀 
https://github.com/DroidKaigi/conference-app-2020/commit/70f7d25873760918b339a90d659e7f30e58db55c

Reply:
Also, besides whether it is good or bad, I would like to challenge this DroidKaigi to make one UiModel.

Reply:
OK, let's challenge the one UiModel!
70f7d25 is enough to simple too for me.